File: frontend/src/services/productService.js
import apiClient from './apiClient';

const productService = {
  getAllProducts: () => apiClient.get('/products').then((res) => res.data),

  // params: { keyword, category, minPrice, maxPrice, inStock, page, size, sortBy, sortDir }
  searchProducts: (params = {}) =>
    apiClient.get('/products/search', { params }).then((res) => res.data),

  getCategories: () => apiClient.get('/products/categories').then((res) => res.data),

  getProductById: (id) => apiClient.get(`/products/${id}`).then((res) => res.data),

  addProduct: (payload) => apiClient.post('/products', payload).then((res) => res.data),

  updateProduct: (id, payload) =>
    apiClient.put(`/products/${id}`, payload).then((res) => res.data),

  deleteProduct: (id) => apiClient.delete(`/products/${id}`).then((res) => res.data),
};

export default productService;

File: seedsanskriti_backend/src/main/java/com/seedsanskriti/service/ProductService.java
package com.seedsanskriti.service;

import java.util.List;

import org.springframework.security.core.Authentication;

import com.seedsanskriti.dto.AddProductRequest;
import com.seedsanskriti.dto.PagedResponse;
import com.seedsanskriti.dto.ProductResponse;
import com.seedsanskriti.dto.UpdateProductRequest;
import com.seedsanskriti.enums.Category;

public interface ProductService {
	
	 String addProduct(AddProductRequest request,
             Authentication authentication);
	 
	 List<ProductResponse> getAllProducts();
	 
	 ProductResponse getProductById(Long id);
	 
	 String updateProduct(Long productId,
             UpdateProductRequest request,
             Authentication authentication);
	 
	 String deleteProduct(Long productId, Authentication authentication);

	 List<ProductResponse> getSupplierProducts(Authentication authentication);

	 PagedResponse<ProductResponse> searchProducts(
			 String keyword,
			 Category category,
			 Double minPrice,
			 Double maxPrice,
			 Boolean inStockOnly,
			 int page,
			 int size,
			 String sortBy,
			 String sortDir);

	 List<Category> getCategories();

}
